include/zephyr/kernel.h

```diff
@@ -122,6 +122,33 @@ typedef void (*k_thread_user_cb_t)(const struct k_thread *thread,
  */
 void k_thread_foreach(k_thread_user_cb_t user_cb, void *user_data);
 
+/**
+ * @brief Iterate over all the threads in running on current cpu.
+ *
+ * This function is does otherwise the same thing as k_thread_foreach(),
+ * but it only loops through the threads running on current core. If
+ * CONFIG_SMP is not defined the implementation this is the same as
+ * k_thread_foreach().
+ *
+ * @param user_cb Pointer to the user callback function.
+ * @param user_data Pointer to user data.
+ *
+ * @note @kconfig{CONFIG_THREAD_MONITOR} must be set for this function
+ * to be effective.
+ * @note This API uses @ref k_spin_lock to protect the _kernel.threads
+ * list which means creation of new threads and terminations of existing
+ * threads are blocked until this API returns.
+ */
+#ifdef CONFIG_SMP
+void k_thread_foreach_my_core(k_thread_user_cb_t user_cb, void *user_data);
```

Comment:
Instead of using 'xxx_my_core', it might be more consistent to use 'xxx_current_cpu'.